applications/GeoMechanicsApplication/custom_constitutive/thermal_dispersion_2D_law.cpp

```diff
@@ -0,0 +1,67 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//
+
+#include "custom_constitutive/thermal_dispersion_2D_law.hpp"
+#include "geo_mechanics_application_variables.h"
+#include <iostream>
+
+namespace Kratos {
+
+ConstitutiveLaw::Pointer GeoThermalDispersion2DLaw::Clone() const
+{
+    return Kratos::make_shared<GeoThermalDispersion2DLaw>(*this);
+}
+
+SizeType GeoThermalDispersion2DLaw::WorkingSpaceDimension()
+{
+    constexpr SizeType space_dimension = 2;
+    return space_dimension;
+}
+
+void GeoThermalDispersion2DLaw::CalculateThermalDispersionMatrix(Matrix& rThermalDispersionMatrix,
+                                                                 const Properties& rProp)
+{
+    KRATOS_TRY
+
+    const double cWater = rProp[POROSITY] * rProp[SATURATION];
```

Comment:
For the SATURATION, @WPK4FEM proposed to investate if we could re-use the retention laws that are already there instead of having this new entry in the MaterialParameters.json. I think it would be a good idea to discuss this in office.